coalib/results/Diff.py

```diff
@@ -70,6 +72,55 @@ def from_string_arrays(cls, file_array_1, file_array_2, rename=False):
         return result
 
     @classmethod
+    def from_unified_diff(cls, unified_diff, original_file):
+        """
+        Creates a ``Diff`` object from given unified diff.
+
+        If this ``Diff`` is applied to the original file, the corrected
+        file will be created.
+
+        :param unified_diff:  Unified diff string.
+        :param original_file: The contents of the original file
+                              (line-splitted).
+        """
+        patch_set = PatchSet(unified_diff.splitlines())
+
+        diff = Diff(original_file)
+
+        for file in patch_set:
+            for hunk in file:
+                file_line = hunk.source_start
+                i = 0
+                while i < len(hunk):
+                    line = hunk[i]
+                    source_code = str(line)[1:]
+
+                    if line.is_added:
+                        add_set = []
+                        # Add all the consecutive '+' lines at once
+                        while(i < len(hunk) and hunk[i].is_added):
+                            source_code = str(hunk[i])[1:]
+                            add_set.append(source_code)
+                            i += 1
+                        diff.add_lines(file_line-1, add_set)
+
+                    elif line.is_removed:
+                        if source_code != original_file[file_line-1].strip("\n"):
```

Comment:
Had to add `strip("\n")` because newlines are stripped of by `unidiff` but the `original_file` contains newlines.
(even 
```py
 PatchSet(unified_diff.splitlines(keepends=True))
``` 
doesn't help)